### PR TITLE
Shellgit

### PR DESCRIPTION
**Pull request by chris-steele**
Replaced `nodegit` in favour of shell commands. Tested in Node v8 and v10. Addresses persistent problems with authentication in some environments (particularly Linux and macOS).

### CODE REVIEW

**Comment by oliverfoster** [comment]
ahhh, the dream, no more nodegit compilation / installation / version issues :dreamyeyes:

Windows 10, git bash, command prompt, node 8.9.4:

image

**Comment by chris-steele** [comment]
Awesome thanks @oliverfoster. So that's Windows and Linux covered, let's see if we can get a macOS citizen; @guywillis , @kirsty-hames have you got 5 mins to test this? Easiest way is probably to do `npm install -g adapt-cli@2.1.4` and then do `adapt devinstall`. If you want to go back to the latest release you can then do `npm install -g adapt@latest`

**Comment by moloko** [comment]
`adapt create course` worked fine for me but `adapt devinstall fails` I'm afraid :-(

node-10

node-8

File: lib/commands/devinstall.js [hunk 1]
```diff
@@ -82,17 +81,21 @@ module.exports = function (dependencies) {
                 return nodegit.Repository.open(localPath);
```
Comment (moloko) [inline lib/commands/devinstall.js]: there's a reference to `nodegit` here still...

**Comment by moloko** [comment]
Testing on macOS 10.14
mac-mojave

**Comment by oliverfoster** [comment]
Does it work on windows without spaces in the folder name?

**Comment by oliverfoster** [comment]
Can you try again and screen print?

**Comment by moloko** [comment]
sure... 
2018-11-07 20_17_38-cmder

**Comment by chris-steele** [comment]
@oliverfoster, @moloko thanks for the support on this; much appreciated. I am working on resolving this issue today.

**Comment by chris-steele** [comment]
@moloko, @oliverfoster should be fixed now. I've tested on Linux and Windows (on the latter with `cmd`, `powershell` and `cmder`) and it's working fine. The problem was that the code didn't switch to the framework directory and hence the bower commands couldn't see .bowerrc. This meant none of the plugins were found. Right now I can't understand how it ever worked. Despite this, before I made this change it worked for me in Linux and on Windows: when I didn't use paths with spaces. I have spent a while resolving this so don't really want to find out how it ever worked, but possible `bower-config/lib/util/rc.js` holds the answer as it seems to be doing path resolution using various locations. I guess it picks up the correct .bowerrc somewhere. Thanks for reading.

**Comment by chris-steele** [comment]
@moloko if you wouldn't mind doing a quick macOS test, @guywillis too if you can...

**Comment by guywillis** [comment]
Seems to work fine on MacOS in v8 and v10 of node

screen shot 2018-11-09 at 09 50 25
screen shot 2018-11-09 at 09 51 41

**Pull request merged**